DOCS-11175 Dropbox 1.0 (#1520)
mulesoft/docs-connectors: dropbox/1.0/modules/ROOT/pages/dropbox-connector-reference.adoc

= Dropbox Connector 1.0 Reference
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

== Configurations
---

> [[Config]]

Reviewer: You can delete 9&10 since there is only one configuration.

|Name | String | Name for this configuration. Connectors reference the configuration with this name. | | x
| Connection a| <<Config_Oauth2, Oauth2 Connection Provider>>
 | Connection types for this configuration. | | x
| Response Timeout a| Number |  How long Mule waits for a response to complete before the response times out. |  1 |
| Response Timeout Unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS
** DAYS |  Time unit for the *Response Timeout* field. |  MINUTES |
| Streaming Type a| Enumeration, one of:

** AUTO
** ALWAYS
** NEVER | How the connector streams request content to the remote service:

* `AUTO` (default)

Connector automatically uses the best streaming strategy based on the request content.

* `ALWAYS`

Connector always streams the request content.

* `NEVER`

Connector does not stream the request content. |  AUTO |
| Name a| String |  Identifier of this element used to reference it in other components. |  | x
| Expiration Policy a| <<ExpirationPolicy>> |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before Mule considers it eligible for expiration. |  |
|===

> ==== Connection Types

Reviewer: Make Connection Types H3

[[Config_Oauth2]]

> ===== Oauth2 Connection Provider

Reviewer: H4

| Connection Idle Timeout Unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS
** DAYS |  Time unit for the *Connection Idle Timeout* field. |  SECONDS |
| Proxy Config a| <<Proxy>> |  Configures a proxy for outbound connections. |  |
| Stream Response a| Boolean |  If this value is `true`, Mule streams received responses. |  false |
| Response Buffer Size a| Number |  Size of the buffer that stores the HTTP response, in bytes. |  -1 |
| Token Access Type a| String |  If this parameter is set to `offline`, then the access token payload contains a long-lived refresh_token. |  offline |
| Protocol a| Enumeration, one of:

** HTTP
** HTTPS |  Protocol to use for communication. |  HTTP |
| TLS Configuration a| <<Tls>> | Configures TLS. If using the HTTPS protocol, you must configure TLS.  |  |
| Reconnection a| <<Reconnection>> |  When the application is deployed, a connectivity test is performed on all connectors. If set to `true`, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  |

> | Consumer Key a| String |  OAuth consumerKey as registered with the service provider. |  | x

Reviewer: ```suggestion
| Consumer Key a| String |  OAuth consumer key as registered with the service provider. |  | x
```

> | Consumer Secret a| String |  OAuth consumerSecret as registered with the service provider. |  | x

Reviewer: ```suggestion
| Consumer Secret a| String |  OAuth consumer secret as registered with the service provider. |  | x
```

| Authorization Url a| String |  Service provider's authorization endpoint URL. |  +https://www.dropbox.com/oauth2/authorize+ |

> | Access Token Url a| String |  Service provider's accessToken endpoint URL. |  +https://api.dropboxapi.com/oauth2/token+ |

Reviewer: ```suggestion
| Access Token Url a| String |  Service provider's access token endpoint URL. |  +https://api.dropboxapi.com/oauth2/token+ |
```

| Resource Owner Id a| String |  Resource owner ID to use with the authorization code grant type. |  |
| Before a| String |  Name of a flow to execute immediately before starting the OAuth dance. |  |
| After a| String |  Name of a flow to execute immediately after receiving an access token. |  |
| Listener Config a| String |  A reference to an `<http:listener-config />` to use to create the listener that receives the access token callback endpoint. |  | x
| Callback Path a| String |  Path of the access token callback endpoint. |  | x
| Authorize Path a| String |  Path of the local HTTP endpoint that triggers the OAuth dance. |  | x
| External Callback Url a| String |  If the callback endpoint is behind a proxy or must be accessed through a non direct URL, use this parameter to tell the OAuth provider the URL to use to access the callback. |  |
| Object Store a| String |  A reference to the object store to use to store each resource owner ID's data. If not specified, Mule automatically provisions the default one. |  |
|===

> == Associated Operations

Reviewer: This should be H2 and just say Operations

* <<CreateSharingListFileMembersContinue>>
* <<CreateSharingListFolderMembers>>
* <<CreateSharingListFolderMembersContinue>>
* <<CreateSharingListFolders>>
* <<CreateSharingListFoldersContinue>>
* <<CreateSharingListMountableFolders>>
* <<CreateSharingListMountableFoldersContinue>>
* <<CreateSharingListReceivedFiles>>
* <<CreateSharingListReceivedFilesContinue>>
* <<CreateSharingListSharedLinks>>
* <<CreateSharingModifySharedLinkSettings>>
* <<CreateSharingMountFolder>>
* <<CreateSharingRelinquishFileMembership>>
* <<CreateSharingRelinquishFolderMembership>>
* <<CreateSharingRemoveFileMember2>>
* <<CreateSharingRemoveFolderMember>>
* <<CreateSharingRevokeSharedLink>>
* <<CreateSharingSetAccessInheritance>>
* <<CreateSharingShareFolder>>
* <<CreateSharingTransferFolder>>
* <<CreateSharingUnmountFolder>>
* <<CreateSharingUnshareFile>>
* <<CreateSharingUnshareFolder>>
* <<CreateSharingUpdateFileMember>>
* <<CreateSharingUpdateFolderMember>>
* <<CreateSharingUpdateFolderPolicy>>
* <<CreateUsersFeaturesGetValues>>
* <<CreateUsersGetAccount>>
* <<CreateUsersGetAccountBatch>>
* <<CreateUsersGetCurrentAccount>>
* <<CreateUsersGetSpaceUsage>>
* <<Unauthorize>>

> == Operations

Reviewer: Remove this header

* `ALWAYS`

Connector always streams the request content.

* `NEVER`

Connector does not stream the request content. |  |
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
|===

=== Output

[%autowidth.spread]
|===
|Type |Any
| Attributes Type a| <<HttpResponseAttributes>>
|===

=== For Configurations

* <<Config>>

=== Throws

* DROPBOX-USERV2:BAD_REQUEST
* DROPBOX-USERV2:CLIENT_ERROR
* DROPBOX-USERV2:CONNECTIVITY
* DROPBOX-USERV2:INTERNAL_SERVER_ERROR
* DROPBOX-USERV2:NOT_ACCEPTABLE
* DROPBOX-USERV2:NOT_FOUND
* DROPBOX-USERV2:RETRY_EXHAUSTED
* DROPBOX-USERV2:SERVER_ERROR
* DROPBOX-USERV2:SERVICE_UNAVAILABLE
* DROPBOX-USERV2:TIMEOUT
* DROPBOX-USERV2:TOO_MANY_REQUESTS
* DROPBOX-USERV2:UNAUTHORIZED
* DROPBOX-USERV2:UNSUPPORTED_MEDIA_TYPE

[[CreateFileRequestsGet]]
== File Requests Get
`<dropbox-userv2:create-file-requests-get>`

> Returns the specified file request. This operation makes an HTTP POST request to the `/file_requests/get endpoint`.

Reviewer: ```suggestion
Returns the specified file request. This operation makes an HTTP POST request to the `/file_requests/get` endpoint.
```

* `ALWAYS`

Connector always streams the request content.

* `NEVER`

Connector does not stream the request content. |  |
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
|===

=== Output

[%autowidth.spread]
|===
|Type |* Any
* Any
| Attributes Type a| <<HttpResponseAttributes>>
|===

=== For Configurations

* <<Config>>

=== Throws

* DROPBOX-USERV2:BAD_REQUEST
* DROPBOX-USERV2:CLIENT_ERROR
* DROPBOX-USERV2:CONNECTIVITY
* DROPBOX-USERV2:INTERNAL_SERVER_ERROR
* DROPBOX-USERV2:NOT_ACCEPTABLE
* DROPBOX-USERV2:NOT_FOUND
* DROPBOX-USERV2:RETRY_EXHAUSTED
* DROPBOX-USERV2:SERVER_ERROR
* DROPBOX-USERV2:SERVICE_UNAVAILABLE
* DROPBOX-USERV2:TIMEOUT
* DROPBOX-USERV2:TOO_MANY_REQUESTS
* DROPBOX-USERV2:UNAUTHORIZED
* DROPBOX-USERV2:UNSUPPORTED_MEDIA_TYPE

[[CreateFilesCopyBatchV2]]
== Files Copy Batch V2
`<dropbox-userv2:create-files-copy-batch-v2>`

> Copies multiple files or folders to different locations at once in the user's Dropbox. This route will replace copy_batch:1. The main difference is that this route will return a status for each entry, while copy_batch:1 raises failure if any entry fails. This route will either finish synchronously, or return a job ID and do the asynchronous copy job in background. Use copy_batch/check:2 to check the job status. This operation makes an HTTP POST request to the `/files/copy_batch_v2` endpoint.

Reviewer: ```suggestion
Copies multiple files or folders simultaneously to different locations in the user's Dropbox. This route replaces `copy_batch:1`. The main difference is that this route returns a status for each entry, while `copy_batch:1` raises a failure if any entry fails. This route will either finish synchronously or return a job ID and perform the asynchronous copy job in the background. Use `copy_batch/check:2` to check the job status. This operation makes an HTTP POST request to the `/files/copy_batch_v2` endpoint.
```

* `ALWAYS`

Connector always streams the request content.

* `NEVER`

Connector does not stream the request content. |  |
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
|===

=== Output

[%autowidth.spread]
|===
|Type |Any
| Attributes Type a| <<HttpResponseAttributes>>
|===

=== For Configurations

* <<Config>>

=== Throws

* DROPBOX-USERV2:BAD_REQUEST
* DROPBOX-USERV2:CLIENT_ERROR
* DROPBOX-USERV2:CONNECTIVITY
* DROPBOX-USERV2:INTERNAL_SERVER_ERROR
* DROPBOX-USERV2:NOT_ACCEPTABLE
* DROPBOX-USERV2:NOT_FOUND
* DROPBOX-USERV2:RETRY_EXHAUSTED
* DROPBOX-USERV2:SERVER_ERROR
* DROPBOX-USERV2:SERVICE_UNAVAILABLE
* DROPBOX-USERV2:TIMEOUT
* DROPBOX-USERV2:TOO_MANY_REQUESTS
* DROPBOX-USERV2:UNAUTHORIZED
* DROPBOX-USERV2:UNSUPPORTED_MEDIA_TYPE

[[CreateFilesCreateFolderBatch]]
== Files Create Folder Batch
`<dropbox-userv2:create-files-create-folder-batch>`

> Creates multiple folders at once. This route is asynchronous for large batches, which returns a job ID immediately and runs the create folder batch asynchronously. Otherwise, it creates the folders and returns the result synchronously for smaller inputs. You can force asynchronous behavior by using the `CreateFolderBatchArg.force_async` flag. Use `create_folder_batch/check` to check the job status. This operation makes an HTTP POST request to the `/files/create_folder_batch` endpoint.

Reviewer: ```suggestion
Creates multiple folders simultaneously. This route is asynchronous for large batches and returns a job ID immediately, then runs the create folder batch asynchronously. Otherwise, it creates the folders and returns the result synchronously for smaller inputs. You can force asynchronous behavior by using the `CreateFolderBatchArg.force_async` flag. Use `create_folder_batch/check` to check the job status. This operation makes an HTTP POST request to the `/files/create_folder_batch` endpoint.
```

* `ALWAYS`

Connector always streams the request content.

* `NEVER`

Connector does not stream the request content. |  |
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
|===

=== Output

[%autowidth.spread]
|===
|Type |Any
| Attributes Type a| <<HttpResponseAttributes>>
|===

=== For Configurations

* <<Config>>

=== Throws

* DROPBOX-USERV2:BAD_REQUEST
* DROPBOX-USERV2:CLIENT_ERROR
* DROPBOX-USERV2:CONNECTIVITY
* DROPBOX-USERV2:INTERNAL_SERVER_ERROR
* DROPBOX-USERV2:NOT_ACCEPTABLE
* DROPBOX-USERV2:NOT_FOUND
* DROPBOX-USERV2:RETRY_EXHAUSTED
* DROPBOX-USERV2:SERVER_ERROR
* DROPBOX-USERV2:SERVICE_UNAVAILABLE
* DROPBOX-USERV2:TIMEOUT
* DROPBOX-USERV2:TOO_MANY_REQUESTS
* DROPBOX-USERV2:UNAUTHORIZED
* DROPBOX-USERV2:UNSUPPORTED_MEDIA_TYPE

[[CreateFilesDeleteBatch]]
== Files Delete Batch
`<dropbox-userv2:create-files-delete-batch>`

> Delete multiple files/folders at once. This route is asynchronous, which returns a job ID immediately and runs the delete batch asynchronously. Use `delete_batch/check` to check the job status. This operation makes an HTTP POST request to the /files/delete_batch endpoint

Reviewer: ```suggestion
Deletes multiple files and folders simultaneously. This route is asynchronous and returns a job ID immediately, then runs the `delete_batch` operation asynchronously. Use `delete_batch/check` to check the job status. This operation makes an HTTP POST request to the `/files/delete_batch` endpoint.
```

* `ALWAYS`

Connector always streams the request content.

* `NEVER`

Connector does not stream the request content. |  |
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
|===

=== Output

[%autowidth.spread]
|===
|Type |* Any
* Any
* Any
| Attributes Type a| <<HttpResponseAttributes>>
|===

=== For Configurations

* <<Config>>

=== Throws

* DROPBOX-USERV2:BAD_REQUEST
* DROPBOX-USERV2:CLIENT_ERROR
* DROPBOX-USERV2:CONNECTIVITY
* DROPBOX-USERV2:INTERNAL_SERVER_ERROR
* DROPBOX-USERV2:NOT_ACCEPTABLE
* DROPBOX-USERV2:NOT_FOUND
* DROPBOX-USERV2:RETRY_EXHAUSTED
* DROPBOX-USERV2:SERVER_ERROR
* DROPBOX-USERV2:SERVICE_UNAVAILABLE
* DROPBOX-USERV2:TIMEOUT
* DROPBOX-USERV2:TOO_MANY_REQUESTS
* DROPBOX-USERV2:UNAUTHORIZED
* DROPBOX-USERV2:UNSUPPORTED_MEDIA_TYPE

[[CreateFilesDeleteBatchCheck]]
== Files Delete Batch Check
`<dropbox-userv2:create-files-delete-batch-check>`

> Returns the status of an asynchronous job for `delete_batch`. If success, it returns list of result for each entry. This operation makes an HTTP POST request to the /files/delete_batch/check endpoint

Reviewer: ```suggestion
Returns the status of an asynchronous job for `delete_batch`. If it's successful, it returns a list of results for each entry. This operation makes an HTTP POST request to the `/files/delete_batch/check` endpoint.
```

* `ALWAYS`

Connector always streams the request content.

* `NEVER`

Connector does not stream the request content. |  |
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
|===

=== Output

[%autowidth.spread]
|===
|Type |String
| Attributes Type a| <<HttpResponseAttributes>>
|===

=== For Configurations

* <<Config>>

=== Throws

* DROPBOX-USERV2:BAD_REQUEST
* DROPBOX-USERV2:CLIENT_ERROR
* DROPBOX-USERV2:CONNECTIVITY
* DROPBOX-USERV2:INTERNAL_SERVER_ERROR
* DROPBOX-USERV2:NOT_ACCEPTABLE
* DROPBOX-USERV2:NOT_FOUND
* DROPBOX-USERV2:RETRY_EXHAUSTED
* DROPBOX-USERV2:SERVER_ERROR
* DROPBOX-USERV2:SERVICE_UNAVAILABLE
* DROPBOX-USERV2:TIMEOUT
* DROPBOX-USERV2:TOO_MANY_REQUESTS
* DROPBOX-USERV2:UNAUTHORIZED
* DROPBOX-USERV2:UNSUPPORTED_MEDIA_TYPE

[[CreateFilesGetTemporaryLink]]
== Files Get Temporary Link
`<dropbox-userv2:create-files-get-temporary-link>`

> Gets a temporary link to stream content of a file. This link expires in four hours and after that you will get a 410 Gone. The content-type of the link is determined automatically by the file's mime type. This operation makes an HTTP POST request to the `/files/get_temporary_link` endpoint.

Reviewer: ```suggestion
Gets a temporary link to stream the content of a file. This link expires after four hours, at which time you receive a `410 Gone` message. The `content-type` of the link is determined automatically by the file's MIME type. This operation makes an HTTP POST request to the `/files/get_temporary_link` endpoint.
```

* `ALWAYS`

Connector always streams the request content.

* `NEVER`

Connector does not stream the request content. |  |
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
|===

=== Output

[%autowidth.spread]
|===
|Type |Any
| Attributes Type a| <<HttpResponseAttributes>>
|===

=== For Configurations

* <<Config>>

=== Throws

* DROPBOX-USERV2:BAD_REQUEST
* DROPBOX-USERV2:CLIENT_ERROR
* DROPBOX-USERV2:CONNECTIVITY
* DROPBOX-USERV2:INTERNAL_SERVER_ERROR
* DROPBOX-USERV2:NOT_ACCEPTABLE
* DROPBOX-USERV2:NOT_FOUND
* DROPBOX-USERV2:RETRY_EXHAUSTED
* DROPBOX-USERV2:SERVER_ERROR
* DROPBOX-USERV2:SERVICE_UNAVAILABLE
* DROPBOX-USERV2:TIMEOUT
* DROPBOX-USERV2:TOO_MANY_REQUESTS
* DROPBOX-USERV2:UNAUTHORIZED
* DROPBOX-USERV2:UNSUPPORTED_MEDIA_TYPE

[[CreateFilesListFolderContinue]]
== Files List Folder Continue
`<dropbox-userv2:create-files-list-folder-continue>`

> Once a cursor has been retrieved from `list_folder`, use this to paginate through all the files and retrieve updates to the folder, following the same rules as documented for `list_folder`. This operation makes an HTTP POST request to the `/files/list_folder/continue` endpoint.

Reviewer: ```suggestion
After a cursor is retrieved from `list_folder`, use this to paginate through all the files and retrieve updates to the folder, following the same rules as documented for `list_folder`. This operation makes an HTTP POST request to the `/files/list_folder/continue` endpoint.
```

* `ALWAYS`

Connector always streams the request content.

* `NEVER`

Connector does not stream the request content. |  |
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
|===

=== Output

[%autowidth.spread]
|===
|Type |Any
| Attributes Type a| <<HttpResponseAttributes>>
|===

=== For Configurations

* <<Config>>

=== Throws

* DROPBOX-USERV2:BAD_REQUEST
* DROPBOX-USERV2:CLIENT_ERROR
* DROPBOX-USERV2:CONNECTIVITY
* DROPBOX-USERV2:INTERNAL_SERVER_ERROR
* DROPBOX-USERV2:NOT_ACCEPTABLE
* DROPBOX-USERV2:NOT_FOUND
* DROPBOX-USERV2:RETRY_EXHAUSTED
* DROPBOX-USERV2:SERVER_ERROR
* DROPBOX-USERV2:SERVICE_UNAVAILABLE
* DROPBOX-USERV2:TIMEOUT
* DROPBOX-USERV2:TOO_MANY_REQUESTS
* DROPBOX-USERV2:UNAUTHORIZED
* DROPBOX-USERV2:UNSUPPORTED_MEDIA_TYPE

[[CreateFilesListFolderLongpoll]]
== Files List Folder Longpoll
`<dropbox-userv2:create-files-list-folder-longpoll>`

> A longpoll endpoint to wait for changes on an account. In conjunction with `list_folder/continue`, this call gives you a low-latency way of monitoring an account for file changes. The connection will block until there are changes available or until a timeout occurs. This endpoint is most useful for client-side apps. If you're looking for server-side notifications, refer to https://www.dropbox.com/developers/reference/webhooks[Webhooks Documentation]. This operation makes an HTTP POST request to the `/files/list_folder/longpoll` endpoint.

Reviewer: ```suggestion
A longpoll endpoint that waits for changes on an account. In conjunction with `list_folder/continue`, this call gives you a low-latency way of monitoring an account for file changes. The connection is blocked until there are changes available or until a timeout occurs. This endpoint is most useful for client-side apps. If you're looking for server-side notifications, refer to https://www.dropbox.com/developers/reference/webhooks[Dropbox Webhooks Documentation^]. This operation makes an HTTP POST request to the `/files/list_folder/longpoll` endpoint.
```

* `ALWAYS`

Connector always streams the request content.

* `NEVER`

Connector does not stream the request content. |  |
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
|===

=== Output

[%autowidth.spread]
|===
|Type |* Any
* Any
| Attributes Type a| <<HttpResponseAttributes>>
|===

=== For Configurations

* <<Config>>

=== Throws

* DROPBOX-USERV2:BAD_REQUEST
* DROPBOX-USERV2:CLIENT_ERROR
* DROPBOX-USERV2:CONNECTIVITY
* DROPBOX-USERV2:INTERNAL_SERVER_ERROR
* DROPBOX-USERV2:NOT_ACCEPTABLE
* DROPBOX-USERV2:NOT_FOUND
* DROPBOX-USERV2:RETRY_EXHAUSTED
* DROPBOX-USERV2:SERVER_ERROR
* DROPBOX-USERV2:SERVICE_UNAVAILABLE
* DROPBOX-USERV2:TIMEOUT
* DROPBOX-USERV2:TOO_MANY_REQUESTS
* DROPBOX-USERV2:UNAUTHORIZED
* DROPBOX-USERV2:UNSUPPORTED_MEDIA_TYPE

[[CreateFilesMoveBatchV2]]
== Files Move Batch V2
`<dropbox-userv2:create-files-move-batch-v2>`

> Moves multiple files or folders to different locations at once in the user's Dropbox. This route is 'all or nothing', which means if one entry fails, the whole transaction will abort. This route will return a job ID immediately and do the asynchronous moving job in background. Use `move_batch/check` to check the job status. This operation makes an HTTP POST request to the `/files/move_batch_v2` endpoint.

Reviewer: ```suggestion
Moves multiple files or folders to different locations simultaneously in the user's Dropbox. This route is "all or nothing," which means if one entry fails, the whole transaction will abort. This route returns a job ID immediately and performs the asynchronous moving job in the background. Use `move_batch/check` to check the job status. This operation makes an HTTP POST request to the `/files/move_batch_v2` endpoint.
```

* `ALWAYS`

Connector always streams the request content.

* `NEVER`

Connector does not stream the request content. |  |
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
|===

=== Output

[%autowidth.spread]
|===
|Type |Any
| Attributes Type a| <<HttpResponseAttributes>>
|===

=== For Configurations

* <<Config>>

=== Throws

* DROPBOX-USERV2:BAD_REQUEST
* DROPBOX-USERV2:CLIENT_ERROR
* DROPBOX-USERV2:CONNECTIVITY
* DROPBOX-USERV2:INTERNAL_SERVER_ERROR
* DROPBOX-USERV2:NOT_ACCEPTABLE
* DROPBOX-USERV2:NOT_FOUND
* DROPBOX-USERV2:RETRY_EXHAUSTED
* DROPBOX-USERV2:SERVER_ERROR
* DROPBOX-USERV2:SERVICE_UNAVAILABLE
* DROPBOX-USERV2:TIMEOUT
* DROPBOX-USERV2:TOO_MANY_REQUESTS
* DROPBOX-USERV2:UNAUTHORIZED
* DROPBOX-USERV2:UNSUPPORTED_MEDIA_TYPE

[[CreateFilesPermanentlyDelete]]
== Files Permanently Delete
`<dropbox-userv2:create-files-permanently-delete>`

> Permanently deletes the file or folder at a given path (refer to https://www.dropbox.com/en/help/40). This operation makes an HTTP POST request to the `/files/permanently_delete` endpoint.

Reviewer: ```suggestion
Permanently deletes the file or folder at a given path (refer to https://www.dropbox.com/en/help/40[Dropbox Help^]). This operation makes an HTTP POST request to the `/files/permanently_delete` endpoint.
```

* `ALWAYS`

Connector always streams the request content.

* `NEVER`

Connector does not stream the request content. |  |
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
|===

=== Output

[%autowidth.spread]
|===
|Type |Any
| Attributes Type a| <<HttpResponseAttributes>>
|===

=== For Configurations

* <<Config>>

=== Throws

* DROPBOX-USERV2:BAD_REQUEST
* DROPBOX-USERV2:CLIENT_ERROR
* DROPBOX-USERV2:CONNECTIVITY
* DROPBOX-USERV2:INTERNAL_SERVER_ERROR
* DROPBOX-USERV2:NOT_ACCEPTABLE
* DROPBOX-USERV2:NOT_FOUND
* DROPBOX-USERV2:RETRY_EXHAUSTED
* DROPBOX-USERV2:SERVER_ERROR
* DROPBOX-USERV2:SERVICE_UNAVAILABLE
* DROPBOX-USERV2:TIMEOUT
* DROPBOX-USERV2:TOO_MANY_REQUESTS
* DROPBOX-USERV2:UNAUTHORIZED
* DROPBOX-USERV2:UNSUPPORTED_MEDIA_TYPE

[[CreateFilesSaveUrl]]
== Files Save Url
`<dropbox-userv2:create-files-save-url>`

> Saves a specified URL into a file in the user's Dropbox. If the given path already exists, the file will be renamed to avoid the conflict, for example myfile(1).txt. This operation makes an HTTP POST request to the `/files/save_url` endpoint.

Reviewer: ```suggestion
Saves a specified URL into a file in the user's Dropbox. If the given path already exists, the file is renamed to avoid a conflict, for example, `myfile(1).txt`. This operation makes an HTTP POST request to the `/files/save_url` endpoint.
```

* `ALWAYS`

Connector always streams the request content.

* `NEVER`

Connector does not stream the request content. |  |
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
|===

=== Output

[%autowidth.spread]
|===
|Type |* String
* Any
* Any
| Attributes Type a| <<HttpResponseAttributes>>
|===

=== For Configurations

* <<Config>>

=== Throws

* DROPBOX-USERV2:BAD_REQUEST
* DROPBOX-USERV2:CLIENT_ERROR
* DROPBOX-USERV2:CONNECTIVITY
* DROPBOX-USERV2:INTERNAL_SERVER_ERROR
* DROPBOX-USERV2:NOT_ACCEPTABLE
* DROPBOX-USERV2:NOT_FOUND
* DROPBOX-USERV2:RETRY_EXHAUSTED
* DROPBOX-USERV2:SERVER_ERROR
* DROPBOX-USERV2:SERVICE_UNAVAILABLE
* DROPBOX-USERV2:TIMEOUT
* DROPBOX-USERV2:TOO_MANY_REQUESTS
* DROPBOX-USERV2:UNAUTHORIZED
* DROPBOX-USERV2:UNSUPPORTED_MEDIA_TYPE

[[CreateFilesSearchContinueV2]]
== Files Search Continue V2
`<dropbox-userv2:create-files-search-continue-v2>`

> Fetches the next page of search results returned from search:2. search:2 along with search/continue:2 can be used only to retrieve a maximum of 10,000 matches. Recent changes may not immediately be reflected in the search results due to a short delay in indexing. Duplicate results may be returned across pages. Some results may not be returned. This operation makes an HTTP POST request to the `/files/search/continue_v2` endpoint.

Reviewer: ```suggestion
Fetches the next page of search results returned from `search:2`. You can only use `search:2`, along with `search/continue:2`, to retrieve a maximum of 10,000 matches. Recent changes might not immediately be reflected in the search results due to a short delay in indexing. Duplicate results might be returned across pages, or some results might not be returned. This operation makes an HTTP POST request to the `/files/search/continue_v2` endpoint.
```

Connector always streams the request content.

* `NEVER`

Connector does not stream the request content. |  |
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
|===

=== Output

[%autowidth.spread]
|===
|Type |Any
| Attributes Type a| <<HttpResponseAttributes>>
|===

=== For Configurations

* <<Config>>

=== Throws

* DROPBOX-USERV2:BAD_REQUEST
* DROPBOX-USERV2:CLIENT_ERROR
* DROPBOX-USERV2:CONNECTIVITY
* DROPBOX-USERV2:INTERNAL_SERVER_ERROR
* DROPBOX-USERV2:NOT_ACCEPTABLE
* DROPBOX-USERV2:NOT_FOUND
* DROPBOX-USERV2:RETRY_EXHAUSTED
* DROPBOX-USERV2:SERVER_ERROR
* DROPBOX-USERV2:SERVICE_UNAVAILABLE
* DROPBOX-USERV2:TIMEOUT
* DROPBOX-USERV2:TOO_MANY_REQUESTS
* DROPBOX-USERV2:UNAUTHORIZED
* DROPBOX-USERV2:UNSUPPORTED_MEDIA_TYPE

[[CreateFilesUploadSessionFinishBatch]]
== Files Upload Session Finish Batch
`<dropbox-userv2:create-files-upload-session-finish-batch>`

> This route helps you commit many files at once into a user's Dropbox. Use `upload_session/start` and `upload_session/append_v2` to upload file contents. Uploading many files in parallel can help increase throughput. Once the file contents have been uploaded, instead of calling `upload_session/finish`, use this route to finish all of your upload sessions in a single request. `UploadSessionStartArg.close` or `UploadSessionAppendArg.close` need to be true for the last `upload_session/start` or `upload_session/append_v2` call. The maximum size of a file one can upload to an upload session is 350 GB. This route will return a job_id immediately and do the asynchronous commit job in background. Use `upload_session/finish_batch/check` to check the job status. For the same account, this route should be executed serially, which means you should not start the next job before the current job finishes. You can allow up to 1000 entries in a single request. This operation makes an HTTP POST request to the `/files/upload_session/finish_batch` endpoint.

Reviewer: ```suggestion
This route enables you to commit many files simultaneously into a user's Dropbox. Use `upload_session/start` and `upload_session/append_v2` to upload file contents. Uploading many files in parallel can help increase throughput. After the file contents are uploaded, instead of calling `upload_session/finish`, use this route to finish all of your upload sessions in a single request. `UploadSessionStartArg.close` or `UploadSessionAppendArg.close` must be true for the last `upload_session/start` or `upload_session/append_v2` call. The maximum size of a file you can upload to an upload session is 350 GB. This route returns a job_id immediately and performs the asynchronous commit job in the background. Use `upload_session/finish_batch/check` to check the job status. For the same account, this route should be executed serially, which means you should not start the next job before the current job finishes. You can have up to 1000 entries in a single request. This operation makes an HTTP POST request to the `/files/upload_session/finish_batch` endpoint.
```

Connector always streams the request content.

* `NEVER`

Connector does not stream the request content. |  |
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
|===

=== Output

[%autowidth.spread]
|===
|Type |Any
| Attributes Type a| <<HttpResponseAttributes>>
|===

=== For Configurations

* <<Config>>

=== Throws

* DROPBOX-USERV2:BAD_REQUEST
* DROPBOX-USERV2:CLIENT_ERROR
* DROPBOX-USERV2:CONNECTIVITY
* DROPBOX-USERV2:INTERNAL_SERVER_ERROR
* DROPBOX-USERV2:NOT_ACCEPTABLE
* DROPBOX-USERV2:NOT_FOUND
* DROPBOX-USERV2:RETRY_EXHAUSTED
* DROPBOX-USERV2:SERVER_ERROR
* DROPBOX-USERV2:SERVICE_UNAVAILABLE
* DROPBOX-USERV2:TIMEOUT
* DROPBOX-USERV2:TOO_MANY_REQUESTS
* DROPBOX-USERV2:UNAUTHORIZED
* DROPBOX-USERV2:UNSUPPORTED_MEDIA_TYPE

[[CreateSharingListFolderMembersContinue]]
== Sharing List Folder Members Continue
`<dropbox-userv2:create-sharing-list-folder-members-continue>`

> Once a cursor has been retrieved from `list_folder_members`, use this to paginate through all shared folder members. Apps must have full Dropbox access to use this endpoint. This operation makes an HTTP POST request to the `/sharing/list_folder_members/continue` endpoint.

Reviewer: ```suggestion
After a cursor is retrieved from `list_folder_members`, use this to paginate through all shared folder members. Apps must have full Dropbox access to use this endpoint. This operation makes an HTTP POST request to the `/sharing/list_folder_members/continue` endpoint.
```

* `ALWAYS`

Connector always streams the request content.

* `NEVER`

Connector does not stream the request content. |  |
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
|===

=== Output

[%autowidth.spread]
|===
|Type |Any
| Attributes Type a| <<HttpResponseAttributes>>
|===

=== For Configurations

* <<Config>>

=== Throws

* DROPBOX-USERV2:BAD_REQUEST
* DROPBOX-USERV2:CLIENT_ERROR
* DROPBOX-USERV2:CONNECTIVITY
* DROPBOX-USERV2:INTERNAL_SERVER_ERROR
* DROPBOX-USERV2:NOT_ACCEPTABLE
* DROPBOX-USERV2:NOT_FOUND
* DROPBOX-USERV2:RETRY_EXHAUSTED
* DROPBOX-USERV2:SERVER_ERROR
* DROPBOX-USERV2:SERVICE_UNAVAILABLE
* DROPBOX-USERV2:TIMEOUT
* DROPBOX-USERV2:TOO_MANY_REQUESTS
* DROPBOX-USERV2:UNAUTHORIZED
* DROPBOX-USERV2:UNSUPPORTED_MEDIA_TYPE

[[CreateSharingListFoldersContinue]]
== Sharing List Folders Continue
`<dropbox-userv2:create-sharing-list-folders-continue>`

> Once a cursor has been retrieved from `list_folders`, use this to paginate through all the shared folders. The cursor must come from a previous call to `list_folders` or `list_folders/continue`. Apps must have full Dropbox access to use this endpoint. This operation makes an HTTP POST request to the `/sharing/list_folders/continue` endpoint.

Reviewer: ```suggestion
After a cursor is retrieved from `list_folders`, use this to paginate through all the shared folders. The cursor must come from a previous call to `list_folders` or `list_folders/continue`. Apps must have full Dropbox access to use this endpoint. This operation makes an HTTP POST request to the `/sharing/list_folders/continue` endpoint.
```

=== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | Name of the configuration to use. | | x
| Body a| Any |  Content to use. |  #[payload] |
| Config Ref a| ConfigurationProvider |  Name of the configuration to use to execute this component. |  | x
| Streaming Strategy a| * <<RepeatableInMemoryStream>>
* <<RepeatableFileStoreStream>>
* non-repeatable-stream |  Configures how Mule processes streams. Repeatable streams are the default behavior. |  |
| Custom Query Parameters a| Object | Custom query parameters to include in the request. The specified query parameters are merged with the default query parameters that are specified in the configuration. |  |
| Custom Headers a| Object | Custom headers to include in the request. The specified custom headers are merged with the default headers that are specified in the configuration. |  |
| Response Timeout a| Number |  How long Mule waits for a response to complete before the response times out. |  |
| Response Timeout Unit a| Enumeration, one of:

> ** N

Reviewer: ```suggestion
After a cursor is retrieved from `list_mountable_folders`, use this to paginate through all mountable shared folders. The cursor must come from a previous call to `list_mountable_folders` or `list_mountable_folders/continue`. Apps must have full Dropbox access to use this endpoint. This operation makes an HTTP POST request to the `/sharing/list_mountable_folders/continue` endpoint.
```

> ** N

Reviewer: ```suggestion
Returns a list of all files shared with the current user. Does not include files the user receives through shared folders or unclaimed invitations. This operation makes an HTTP POST request to the `/sharing/list_received_files` endpoint.
```

> ** N

Reviewer: ```suggestion
Gets information about multiple user accounts. You can query a maximum of 300 accounts per request. This operation makes an HTTP POST request to the `/users/get_account_batch` endpoint.
```

> ** N

Reviewer: ```suggestion
Deletes all of the access token information of a given resource owner ID so that it is impossible to execute any operation for that user without repeating the authorization dance.
```
